components/bootloader_support/src/esp_image_format.c

```diff
@@ -190,9 +190,9 @@ static esp_err_t image_load(esp_image_load_mode_t mode, const esp_partition_pos_
 
         data->image_len = end_addr - data->start_addr;
         ESP_LOGV(TAG, "image start 0x%08x end of last section 0x%08x", data->start_addr, end_addr);
-        if (NULL != checksum && !esp_cpu_in_ocd_debug_mode()) {
+        if (NULL != checksum) {
             err = verify_checksum(sha_handle, checksum_word, data);
-            if (err != ESP_OK) {
+            if (err != ESP_OK && !esp_cpu_in_ocd_debug_mode()) {
                 goto err;
             }
         }
```

Comment:
Just realizing it might be better to change it into:

```
#ifdef BOOTLOADER_BUILD
        if (NULL != checksum && !esp_cpu_in_ocd_debug_mode()) {
#else
        if (NULL != checksum) {
#endif
            err = verify_checksum(sha_handle, checksum_word, data);
            if (err != ESP_OK) {
                goto err;
            }
        }
```

instead, and change the other occurrence below into the same format:

```
#ifdef BOOTLOADER_BUILD
            if (sha_handle != NULL && !esp_cpu_in_ocd_debug_mode()) {
#else
            if (sha_handle != NULL) {
#endif
                err = verify_simple_hash(sha_handle, data);
            }
```

The case below I wasn't hitting because I had SECURE_BOOT_CHECK_SIGNATURE set.

 \- Tim